### PR TITLE
[wamr] skip leading paddings in .aot section

### PR DESCRIPTION
**Pull request by lum1n0us**

### CODE REVIEW

**Comment by shukitchan** [comment]
Perhaps we should separate this into 2 PRs?
one for wamr 1.3.0
another to do skip leading padding for aot?

**Comment by PiotrSikora** [comment]
> Perhaps we should separate this into 2 PRs? one for wamr 1.3.0 another to do skip leading padding for aot?

It seems that skipping is done only for AOT, so I don't see a reason for it to be 2 separate PRs.

**Comment by lum1n0us** [comment]
This one is based on #378, and it will only contain "skip aot padding" after rebasing after #378 merged.

File: src/wamr/wamr.cc
Comment (PiotrSikora) [inline src/wamr/wamr.cc:228]: Shouldn't this be changed to `WASM_EXTERNREF` instead of removed completely?

File: bazel/repositories.bzl
Comment (PiotrSikora) [inline bazel/repositories.bzl:211]: We want to make builds even more hermetic, and relying on a system-installed LLVM is definitely not the way to do it. I'd remove this comment.

**Review by PiotrSikora — COMMENTED** [review]
LGTM, sans comments.

File: bazel/repositories.bzl
Comment (lum1n0us) [reply bazel/repositories.bzl:211]: good to know

**Comment by lum1n0us** [comment]
@mpwarres  @martijneken  any update?

File: src/wamr/wamr.cc
Comment (mpwarres) [inline src/wamr/wamr.cc:138]: nit: use a type with explicit size (uint8_t?) and explicit static_cast<...>

**Review by mpwarres — APPROVED** [review]
LGTM mod a few superficial nits. Thanks!

**Review by johnlanni — CHANGES_REQUESTED** [review]

File: src/wamr/wamr.cc
Comment (johnlanni) [inline src/wamr/wamr.cc:132]:
```suggestion
    binary.data = static_cast<char *>(bytecode.data());
```

File: src/wamr/wamr.cc
Comment (johnlanni) [inline src/wamr/wamr.cc:138]:
```suggestion
    uint8_t padding_count = static_cast<uint8_t>(precompiled[0]);
```

File: src/wamr/wamr.cc
Comment (johnlanni) [inline src/wamr/wamr.cc:132]:
```suggestion
    binary.data = static_cast<char *>(bytecode.data());
```

File: src/wamr/wamr.cc
Comment (johnlanni) [inline src/wamr/wamr.cc:142]:
```suggestion
    binary.data = static_cast<char *>(precompiled.data());
```

File: src/wamr/wamr.cc
Comment (mpwarres) [reply src/wamr/wamr.cc:132]: ah, right

**Review by mpwarres — CHANGES_REQUESTED** [review]
Happy to merge this once @johnlanni 's open comments are addressed. Thanks!

**Review by mpwarres — APPROVED** [review]
LG, thanks @lum1n0us !

**Pull request merged**
